test/harry/main/org/apache/cassandra/harry/sut/TokenPlacementModel.java

```diff
@@ -150,12 +150,12 @@ public interface CompareTo<V>
         int compareTo(V v);
     }
 
-    public static void addIfUnique(List<Replica> replicas, Set<Integer> names, Replica replica)
+    private static void addIfUnique(List<Replica> replicas, Set<String> names, Replica replica)
     {
-        if (names.contains(replica.node().idx()))
+        if (names.contains(replica.node().id()))
```

Comment:
in this call path `idx() == 0` for every replica, as they don't *share* the same lookup...  personally a few methods are unsafe with this class, but didn't want to refactor and just needed to make safe, so use `id` which is unique regardless of how you create the type